src/x25519.rs

```diff
@@ -12,13 +12,93 @@
 //! This implements x25519 key exchange as specified by Mike Hamburg
 //! and Adam Langley in [RFC7748](https://tools.ietf.org/html/rfc7748).
 
+use core::ops::Mul;
+
+use clear_on_drop::clear::Clear;
+
 use curve25519_dalek::constants::ED25519_BASEPOINT_TABLE;
 use curve25519_dalek::montgomery::MontgomeryPoint;
 use curve25519_dalek::scalar::Scalar;
 
 use rand_core::RngCore;
 use rand_core::CryptoRng;
 
+/// A DH ephemeral public key.
+#[repr(C)]
+pub struct EphemeralPublic(pub (crate) MontgomeryPoint);
+
+/// A DH ephemeral secret key.
+#[repr(C)]
+#[derive(Default)] // we derive Default in order to use the clear() method in Drop
+pub struct EphemeralSecret(pub (crate) Scalar);
+
+/// Overwrite ephemeral secret key material with null bytes when it goes out of scope.
+impl Drop for EphemeralSecret {
+    fn drop(&mut self) {
+        self.0.clear();
+    }
+}
+
+/// Multiply this `EphemeralPublic` key by a `EphemeralSecret` key.
+impl<'a, 'b> Mul<&'b EphemeralSecret> for &'a EphemeralPublic {
+    type Output = EphemeralPublic;
+
+    fn mul(self, secret: &'b EphemeralSecret) -> EphemeralPublic {
+        EphemeralPublic(self.0 * secret.0)
+    }
+}
```

Comment:
Is there a reason to expose this trait impl if the `diffie_hellman` function exists? My guess is that it's probably not necessary.